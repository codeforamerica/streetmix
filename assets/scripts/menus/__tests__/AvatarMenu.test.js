/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import AvatarMenu from '../AvatarMenu'

const user = {
  id: 'foo'
}

describe('AvatarMenu', () => {
  it('renders user avatar', () => {
    const wrapper = shallow(<AvatarMenu user={user} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders user avatar for admin', () => {
    const user = {
      id: 'foo',
      roles: ['ADMIN']
    }
    const wrapper = shallow(<AvatarMenu user={user} />)
    expect(wrapper.find('.menu-avatar-admin')).toHaveLength(1)
  })

  it('calls click handler', () => {
    const onClick = jest.fn()
    const wrapper = shallow(<AvatarMenu user={user} onClick={onClick} />)
    wrapper.find('button').simulate('click')
    expect(onClick).toHaveBeenCalled()
  })
})
