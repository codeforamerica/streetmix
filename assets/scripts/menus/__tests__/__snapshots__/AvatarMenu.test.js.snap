// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarMenu renders user avatar 1`] = `
<button
  className="menu-attached menu-avatar"
  onClick={[Function]}
>
  <Connect(Avatar)
    userId="foo"
  />
  <span
    className="user-id"
  >
    foo
  </span>
</button>
`;
